adsingestp/parsers/base.py

```diff
@@ -554,10 +554,24 @@ def _detag(self, r, tags_keep):
         # note that parser=lxml is recommended here - if the more stringent lxml-xml is used,
         # the output is slightly different and the code will need to be modified
         newr = self.bsstrtodict(str(r), "lxml")
+
+        # list of all tags in the object
         if newr.find_all():
             tag_list = list(set([x.name for x in newr.find_all()]))
         else:
             tag_list = []
+
+        # list of all processing instructions in the object like <?index>
+        if newr.find_all(string=True):
+            instruction_list = newr.find_all(string=True)
+        else:
+            instruction_list = []
+        for ins in instruction_list:
+            if ins.startswith("index"):
+                # Remove preceding white-space
```

Comment:
this is the fix to remove the unwanted tag + associated preceding whitespace. lmk if this needs to be replaced by the affil fix function